File: code_samples/lua/oop_encapsulation.lua
-- Oop: Encapsulation
-- AI/ML Training Sample

Encapsulation = {}
Encapsulation.__ index = Encapsulation

function Encapsulation:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Encapsulation:process(input)
    self.data = input
end

function Encapsulation:getData()
    return self.data
end

function Encapsulation:validate()
    return #self.data > 0
end

-- Example usage
local instance = Encapsulation:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/dart/oop_encapsulation.dart
// Oop: Encapsulation
// AI/ML Training Sample

class Encapsulation {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Encapsulation();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
